server/server.go

```diff
@@ -598,7 +613,11 @@ func (s *Server) handlePublishWithoutResponse(r *http.Request, v *visitor) (*mes
 			return nil, err
 		}
 	}
-	v.IncrementMessages()
+	if v_billing != nil {
+		v_billing.IncrementMessages()
+	} else {
+		v.IncrementMessages()
+	}
```

Comment:
I would generally opt for replacing `v` early in this function, instead of adding if-then-elses everywhere. 

Case and point: There's a `v.MessageAllowed()` at the top which would be run against the IP-visitor and not the subscriber visitor.

Reply:
Yeah, but in that case, logs and email-allowed need to be if-then-elsed (which makes more sense I guess, since there's fewer of those things). I'll figure out a way to minimize complications.